Objects/longobject.c

```diff
@@ -3830,12 +3780,12 @@ k_mul(PyLongObject *a, PyLongObject *b)
     _Py_DECREF_INT(t1);
     _Py_DECREF_INT(t2);
     if (t3 == NULL) goto fail;
-    assert(Py_SIZE(t3) >= 0);
+    assert(_PyLong_UnsignedDigitCount(t3) >= 0);
```

Comment:
This should really check the sign:
```suggestion
    assert(!_PyLong_IsNegative(t3));
```
Same several times above.